testdata/btf_map_init.c

```diff
@@ -0,0 +1,33 @@
+#include "common.h"
+
+#if __clang_major__ >= 9
+
+int __section("socket/1") tail_1() {
+	return 42;
+}
+
+// Tail call map (program array) initialized with program pointers.
+struct {
+	__uint(type, BPF_MAP_TYPE_PROG_ARRAY);
+	__uint(key_size, sizeof(uint32_t));
+	__uint(max_entries, 2);
+	__array(values, int ());
+} prog_array_init __section(".maps") = {
+	.values = {
+    // Skip index 0 to exercise empty array slots.
+		[1] = &tail_1,
```

Comment:
tailcalls do accept void *ctx, right? Might be a good idea to have a realistic example with the context

Reply:
Yep! The stuff in `testdata/` are contrived examples meant to exercise specific loader code paths, most of the .c files have a disclaimer. :stuck_out_tongue: Forgot it here. We also have an `examples/` folder with more realistic examples, will add something there later.